src/platform/utilities/tests/oauth/crypto.unit.spec.jsx

```diff
@@ -1,21 +1,64 @@
 import { expect } from 'chai';
-import { randomFillSync } from 'crypto';
 import {
   sha256,
   base64UrlEncode,
   generateRandomString,
 } from '../../oauth/crypto';
+import { mockCrypto } from '../../oauth/mockCrypto';
 
 describe('OAuth - Crypto', () => {
+  beforeEach(() => {
+    let hasNodeCrypto = false;
+    try {
+      // eslint-disable-next-line import/no-unresolved
+      require('node:crypto');
+      hasNodeCrypto = true;
+    } catch {
+      hasNodeCrypto = false;
+    }
+
+    if (!hasNodeCrypto) {
+      window.crypto = mockCrypto;
+    }
+  });
```

Comment:
This was the important piece of the puzzle.

Node 22 has `webcrypto` which we rely on for some of our oauth crypto stuff. We try to do a sanity check and see if we can even import `node:crypto` that is solely in Node 22. If its not there we know that we have to use our internal `mockCrypto` because JSDOM and Node 14 don't have it built-in.